janus_core/calculations/md.py

```diff
@@ -555,6 +574,135 @@ def output_files(self) -> None:
             ),
         }
 
+    @property
+    def ramp_temps(self) -> list[float]:
+        """
+        Get list of temperatures in heating/cooling ramp.
+
+        Returns
+        -------
+        list[float]
+            Temperatures in heating/cooling ramp.
+        """
+        ramp_sign = 1 if (self.temp_end - self.temp_start) > 0 else -1
+        ramp_temps = [
+            self.temp_start + ramp_sign * i * self.temp_step
+            for i in range(self.heating_n_temps)
+        ]
+
+        if self.restart:
+            ramp_steps_completed = self.offset // self.heating_steps_per_temp
+            ramp_steps_completed = min(ramp_steps_completed, len(ramp_temps))
+            if isclose(self.temp_start, 0.0):
+                # T~0K steps do not run any MD, so are not included in the offset.
+                # We have to account for that step manually.
+                ramp_steps_completed += 1
+            ramp_temps = ramp_temps[ramp_steps_completed:]
+
+        return ramp_temps
+
+    @property
+    def heating_n_temps(self) -> int:
+        """
+        Get number of temperatures in heating/cooling ramp.
+
+        Returns
+        -------
+        int
+            Number of temperatures in heating/cooling ramp.
+        """
+        # Always include start temperature in ramp, and include end temperature
+        # if separated by an integer number of temperature steps
+        return int(1 + abs(self.temp_end - self.temp_start) // self.temp_step)
+
+    @property
+    def heating_steps_per_temp(self) -> int:
+        """
+        Get number of steps to be run at each temperature in heating/cooling ramp.
+
+        Returns
+        -------
+        int
+            Number of steps to be run at each temperature in heating/cooling ramp.
+        """
+        return int(self.temp_time // self.timestep)
```

Comment:
Shouldn't need to cast after floordiv (`//`)

Reply:
I can get a float (admittedly int-like) out:

```
>>> x = 1.15
>>> x // 1
1.0
```